src/mesh/tdycoremesh.c

```diff
@@ -772,7 +772,405 @@ PetscErrorCode SaveMeshConnectivityInfo(TDy tdy) {
 }
 
 /* -------------------------------------------------------------------------- */
+/// Converts an integer datatype of TDy mesh element in compressed format
+///
+/// @param [in] tdy                  A TDy struct
+/// @param [in] num_elements         Number of elements
+/// @param [in] default_offset_size  Default offset size for elements
+/// @param [in] update_offset        Determines if subelement_offset should be updated
+/// @param [inout] subelement_num    Number of subelements for a given element
+/// @param [inout] subelement_offset Offset of subelements for a given element
+/// @param [inout] subelement_id     Subelement ids
+/// @returns 0                       on success, or a non-zero error code on failure
+PetscErrorCode ConvertMeshElementToCompressedFormatIntegerValues(TDy tdy, PetscInt num_element, PetscInt default_offset_size, PetscInt update_offset,
```

Comment:
Sorry you had to write these conversion functions, @bishtgautam ! I bet we can avoid doing this in a direct constructor for the mesh. But in the meantime, it's great that things are working.